src/component.js

```diff
@@ -91,26 +91,25 @@ Component.prototype.render = Fragment;
 /**
  * @param {import('./internal').Internal} internal The internal to rerender
  */
-function rerender(internal) {
-	if (~internal.flags & MODE_UNMOUNTING && internal.flags & DIRTY_BIT) {
-		const vnode = createVNode(
-			internal.type,
-			internal.props,
-			internal.key, // @TODO we shouldn't need to actually pass these
-			internal.ref, // since the mode flag should bypass key/ref handling
-			0
-		);
-
-		patch(internal, vnode, getParentDom(internal));
-		commitRoot(internal);
-	}
+function renderQueuedInternal(internal) {
+	// Don't render unmounting/unmounted trees:
+	if (internal.flags & MODE_UNMOUNTING) return;
+
+	// Don't render trees already rendered in this pass:
+	if (!(internal.flags & DIRTY_BIT)) return;
+
+	const vnode = createElement(internal.type, internal.props);
+	vnode.props = internal.props;
```

Comment:
Curious as to why this line is needed. Does the diff logic in v11 check for referential equality on props or something?

Reply:
I believe so, yeah. I put it in because the previous code was creating referentially-equal props, but it's possible we don't need it. 

IIRC we have code in the hooks state implementation that is maybe relying on this to differentiate between prop vs state updates.